interface/src/warmup_cooldown_allowance.rs

```diff
@@ -394,8 +412,8 @@ mod test {
                 prop_assert_eq!(float_math_result, 0);
             } else if would_overflow {
                 // In the u128 overflow region, the `f64` implementation is guaranteed to be imprecise.
-                // We only assert that our implementation correctly falls back to account_portion.
-                prop_assert_eq!(integer_math_result, account_portion);
+                // We only assert that the result does not exceed the account's balance
+                prop_assert!(integer_math_result <= account_portion);
```

Comment:
Is it always strictly less than if it'll overflow? Or is it possible to be equal?